apps/molgenis-components/src/components/account/MolgenisSession.vue

```diff
@@ -48,199 +49,70 @@
-      if (schemaSettings && schemaSettings._settings) {
-        this.loadSettings(schemaSettings);
-        this.session.manifest = schemaSettings._manifest;
-      }
-      //set default locale
-      if (this.session.locale === undefined) {
-        //get from cookie
-        const lang = cookies.get("MOLGENIS.locale");
-        if (lang) {
-          this.session.locale = lang;
-        }
-      }
-      //get the map
-      this.loading = false;
-      this.$emit("update:modelValue", this.session);
-    },
-    handleError(reason: IErrorMessage) {
-      this.error = "internal server error " + reason;
-      if (reason?.response?.data?.errors[0]?.message) {
-        this.$emit("error", reason.response.data.errors[0].message);
-      } else {
-        this.$emit("error", this.error);
-      }
-    },
-    parseJson(value: string) {
-      try {
-        return JSON.parse(value);
-      } catch (error) {
-        this.error =
-          "Parsing of settings failed: " + error + ". value: " + value;
-        return null;
-      }
-    },
-    changed() {
-      this.reload();
-      this.showSigninForm = false;
-      this.$emit("update:modelValue", this.session);
-    },
-    closeSigninForm() {
-      this.showSigninForm = false;
-      this.error = null;
-    },
-    closeSignupForm() {
-      this.showSignupForm = false;
-      this.error = null;
-    },
-    async signout() {
-      this.loading = true;
-      this.showSigninForm = false;
-      const data = await request("graphql", `mutation{signout{status}}`).catch(
-        (error: string) => (this.error = "internal server error" + error)
-      );
-      if (data.signout.status === "SUCCESS") {
-        this.session = {};
-      } else {
-        this.error = "sign out failed";
-      }
-      this.loading = false;
-      this.$emit("update:modelValue", this.session);
-      this.reload();
-    },
-  },
-  emits: ["update:modelValue", "error"],
-});
+const locales = computed(() => session.value?.settings?.locales || ["en"]);
+
+function changed() {
+  showSigninForm.value = false;
+  sessionStore.signin();
+}
+
+function closeSigninForm() {
+  showSigninForm.value = false;
+  error.value = "";
+}
+
+function closeSignupForm() {
+  showSignupForm.value = false;
+  error.value = "";
+}
+
+async function signout() {
+  loading.value = true;
+  await sessionStore.signout();
+  showSigninForm.value = false;
+  loading.value = false;
+}
 </script>
 
 <docs>
 <template>
   <div>
-    <MolgenisSession class="bg-primary" v-model="session" graphql="/pet store/graphql"/>
-    <pre>session = {{ session }}</pre>
+    <MolgenisSession class="bg-primary" />
+    <pre>session = {{ JSON.stringify(session, null, 2) }}</pre>
   </div>
 </template>
-<script>
-  export default {
+
+<script lang="ts">
+import { storeToRefs } from "pinia";
+
+export default {
     data() {
       return {
-        session: null
+        session: storeToRefs(this.$sessionStore)
```

Comment:
We need some discussion on when we gonna use 'store' and when we like our components more stupid and then we use props + event hierarchies.